lib/manager/nuget/package-tree.ts

```diff
@@ -0,0 +1,99 @@
+import path from 'path';
+import util from 'util';
+import { glob } from 'glob';
+import Graph from 'graph-data-structure';
+import upath from 'upath';
+import { DOMParser as dom } from 'xmldom';
+import xpath from 'xpath';
+import { GlobalConfig } from '../../config/global';
+import { logger } from '../../logger';
+import { readLocalFile } from '../../util/fs';
+import { regEx } from '../../util/regex';
+
+const globAsync = util.promisify(glob);
+
+// Get all package files at any level of ancestry that depend on `packageFileName`
+export async function getDependentPackageFiles(
+  packageFileName: string
+): Promise<string[]> {
+  const { localDir } = GlobalConfig.get();
+  if (localDir === undefined) {
+    throw new Error('localDir must be set');
+  }
+  const packageFiles = await getAllPackageFiles(localDir);
```

Comment:
Use the package reference instead of searching all files.

Ignore references with variables for now.

Reply:
In .NET, references are defined in the dependent file. So we need to search all files for references to the project file we are operating on to build the reference DAG.